src/simple_http.rs

```diff
@@ -184,54 +204,51 @@ impl Builder {
     }
 
     /// Set the URL of the server to the transport.
-    pub fn url<U: Into<String>>(mut self, url: U) -> Result<Self, Error> {
-        let url = url.into();
-
+    pub fn url(mut self, url: &str) -> Result<Self, Error> {
         // Do some very basic manual URL parsing because the uri/url crates
         // all have unicode-normalization as a dependency and that's broken.
 
-        { // scope for borrowck on url
-            // this clone is here because of 1.29 borrowck
-            // remove this when 1.29 is deprecated
-            let url_cloned = url.clone();
-
-            // We need to get the hostname and the port.
-            // (1) Split scheme
-            let after_scheme = {
-                let mut split = url_cloned.splitn(2, "://");
-                let s = split.next().unwrap();
-                split.next().unwrap_or(s)
-            };
-            // (2) split off path
-            let before_path = after_scheme.splitn(2, "/").next().unwrap();
-            // (3) split off auth part
-            let after_auth = {
-                let mut split = before_path.splitn(2, "@");
-                let s = split.next().unwrap();
-                split.next().unwrap_or(s)
-            };
-            // so now we should have <hostname>:<port> or just <hostname>
-            let mut split = after_auth.split(":");
-            let hostname = split.next().unwrap();
-            let port: u16 = match split.next() {
-                Some(port_str) => match port_str.parse() {
-                    Ok(port) => port,
-                    Err(_) => return Err(Error::InvalidUrl(url)),
-                },
-                None => DEFAULT_PORT,
-            };
-            // make sure we don't have a second colon in this part
-            if split.next().is_some() {
-                return Err(Error::InvalidUrl(url));
+        // We need to get the hostname and the port.
+        // (1) Split scheme
+        let after_scheme = {
+            let mut split = url.splitn(2, "://");
+            let s = split.next().unwrap();
+            split.next().unwrap_or(s)
+        };
+        // (2) split off path
+        let (before_path, path) = {
+            if let Some(slash) = after_scheme.find("/") {
+                (&after_scheme[0..slash], &after_scheme[slash..])
+            } else {
+                (after_scheme, "/")
             }
-
-            self.tp.addr = match format!("{}:{}", hostname, port).to_socket_addrs()?.next() {
-                Some(a) => a,
-                None => return Err(Error::InvalidUrl(url)),
-            };
+        };
+        // (3) split off auth part
+        let after_auth = {
+            let mut split = before_path.splitn(2, "@");
+            let s = split.next().unwrap();
+            split.next().unwrap_or(s)
```

Comment:
We should probably check that the protocol is "http://" rather than just throwing it away. But this is fine for now.

Reply:
it's quite easy to do, I can add that.